ksql-engine/src/main/java/io/confluent/ksql/codegen/SqlToJavaVisitor.java

```diff
@@ -510,19 +511,26 @@ private String visitBooleanComparisonExpression(ComparisonExpression.Type type)
       if (!schemaField.isPresent()) {
         throw new KsqlException("Field not found: " + arrayBaseName);
       }
-      if (schemaField.get().schema().type() == Schema.Type.ARRAY) {
+      final Schema internalSchema = schemaField.get().schema();
+      if (internalSchema.type() == Schema.Type.ARRAY) {
         return new Pair<>(
-            process(node.getBase(), unmangleNames).getLeft() + "[(int)("
-            + process(node.getIndex(), unmangleNames).getLeft() + ")]",
-            schema
+            String.format("((%s) ((%s)%s).get((int)(%s)))",
+                          SchemaUtil.getJavaType(internalSchema.valueSchema()).getSimpleName(),
```

Comment:
extract `SchemaUtil.getJavaType(...)` into a variable outside of the `if` and then you can use it in the 3 places in this method

Reply:
For the value type we still need to call `valueSchema()` for the internalSchema.